508 Compliance for collections/georef/batchgeoreftool.php

RamisaZaman opened the pull request:
# Pull Request Checklist:
508 Compliance for collections/georef/batchgeoreftool.php, Added a few Language Tags in English, Spanish, and French Language.
# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [x] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [x] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [x] There are no linter errors
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [x] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [x] Comment which GitHub issue(s), if any does this PR address
- [x] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [x] It is the code author's responsibility to merge their own pull request after it has been approved
- [x] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option

- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [x] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

RamisaZaman committed:
508 Compliance for collections/georef/batchgeoreftool.php

Atticus29 committed:
add null coalescers to trim calls

Atticus29 committed:
remove deprecated attributes from table and fix spacing and trim null coalescing in collections/georef/batchgeoreftool.php

Atticus29 commented on collections/georef/batchgeoreftool.php:
```diff
@@ -383,7 +399,8 @@
 							</div>
 							<div style="margin:15px;">
-								<table>
+								<table class='xdebug-error xe-deprecated' dir='ltr' style='border: 1px solid black;'>
```
This whole table needs to be replaced with grid-like elements per [our discussion on slack the other day](https://symbiota-support.slack.com/archives/D06E4HD96JF/p1708730334701399?thread_ts=1708627799.608139&cid=D06E4HD96JF).

Atticus29 replied:
Also, `xdebug-error` classes shouldn't be added in. Their presence (presumably in the Total Validator Pro [TVP from now on] pdf I sent you?) just means that the debugger found an issue and rendered a new table in the DOM. Kinda tricky. In this case, this table made by the debugger probably came from deprecation warnings with the `trim()` method that I fixed in [this commit](https://github.com/BioKIC/Symbiota/pull/962/commits/5a2892cfd6d259ef050d658394c9f9ccfa7cf4ee).

Atticus29 reviewed (changes requested):
The `<table>` in the form still needs to be replaced.

I'm also seeing the following issues still in TVP:

<img width="1552" alt="Screen Shot 2024-02-26 at 2 39 16 PM" src="https://github.com/BioKIC/Symbiota/assets/2775448/412cddd5-b0c0-4c99-9d08-2d3d606de273">

My recommendation would be to make the aria-labels unique. So, they get their own translations with slightly different wording.

RamisaZaman committed:
Fixed The Table

I fixed the table according to the information you provided on slack. But I am seeing a little issue with the Longitude (W  = input) where the "=" is not in place.

RamisaZaman commented on collections/georef/batchgeoreftool.php:
The Table format has changed after replacing the td's with div's. Should I do it like this:         
div class="gridlike-form-cell"

Atticus29 replied:
Hi, @RamisaZaman ! `class="gridlike-form-cell"` is not needed. It looks like you have resolved the table issues for the most part, although TVP doesn't seem to like the `colspan` attribute. Here's the latest TVP run results:

[tvp_batchgeoref.pdf](https://github.com/BioKIC/Symbiota/files/14423717/tvp_batchgeoref.pdf)

RamisaZaman committed:
Fixed Table and aria-label

Atticus29 committed:
add skip nav link

Atticus29 reviewed (changes requested):
Hi, @RamisaZaman !

I went ahead and decided not to include a header on this page and instead just add a skip nav link manually. Check out how that was implemented [here](https://github.com/BioKIC/Symbiota/pull/962/commits/c204f379796ede4ca0835b7d2d6c7a2875428b96).

I'm noticing that if you use a keyboard to navigate the query form, the keyboard gets to the "Generate List" button before it gets to the Locality Term input. 

Can you look into how to reverse the order of those and implement the fix?

I think you might start to get somewhere with the `tabindex` attribute.

RamisaZaman committed:
Implementation Fix

Implemented a reversal of the order for the Locality term and Generate List functionalities. Attempted to address the issue using the tabindex attribute, but found that while navigating through tabs, the Locality term was placed before the Home button.

RamisaZaman commented:
Hi @Atticus29,
I observed that while navigating with the keyboard, the ADD option is bypassed unless it is manually clicked on.

Atticus29 reviewed (approved):
Changes look good, and the tab order makes sense now! Thanks!
